script.js

```diff
@@ -58,43 +58,39 @@ let selected = "";
 // Set test result variable
 let testResult = 0;
 
+// Getting the question
+const question = document.getElementById("question");
+// Getting the options
+const op1 = document.getElementById('op1');
+const op2 = document.getElementById('op2');
+
+// Show selection for op1
+op1.addEventListener("click", () => {
+  op1.style.backgroundColor = "lightgoldenrodyellow";
+  selected = op1.value;
+})
+
+// Show selection for op2
+op2.addEventListener("click", () => {
+  op2.style.backgroundColor = "lightgoldenrodyellow";
+  selected = op2.value;
+})
+
```

Comment:
I extracted some code from the `iterate` function.

Thing is, in the previous version of the `iterate` function, each time it fired, it added another event listener to the buttons. It also kept searching the DOM tree for elements.

After I extracted `const` declarations, the DOM tree is searched only 1 time and the `const`s are in the scope anyway. Also the `.addEventListener` calls were moved out of the function so that they execute only 1 time, because they create functions that will be executed every time the user clicks a button!